File: frontend/products-ui/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  colour: string;
  price: number;
  createdAtUtc: string;
}

export interface CreateProductRequest {
  name: string;
  colour: string;
  price: number;
}

export interface LoginResponse {
  accessToken: string;
  expiresAtUtc: string;
}

export interface HealthResponse {
  status: string;
  timestampUtc: string;
}

File: src/ProductsService.Domain/Entities/Product.cs
namespace ProductsService.Domain.Entities;

public class Product
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public required string Colour { get; set; }
    public decimal Price { get; set; }
    public DateTime CreatedAtUtc { get; set; }
}
